extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -301,4 +302,59 @@ export class RemoteQueriesManager extends DisposableObject {
       queryItem.status = QueryStatus.Failed;
     }
   }
+
+  public async exportVariantAnalysisResults(): Promise<void> {
+    const queryId = this.interfaceManager.getCurrentQueryId();
+    const queryHistoryItem = this.qhm.getCurrentQueryHistoryItem();
+
+    if (!queryId || !queryHistoryItem || !queryHistoryItem.completed || queryHistoryItem.t !== 'remote') {
+      throw new Error('No variant analysis results currently open. To open results, click an item in the query history view.');
+    }
+
+    void this.logger.log(`Exporting variant analysis results for query: ${queryId}`);
+    const query = queryHistoryItem.remoteQuery;
+    const analysesResults = this.analysesResultsManager.getAnalysesResults(queryId);
+
+    const credentials = await Credentials.initialize(this.ctx);
+
+    const gistOption = {
+      label: '$(ports-open-browser-icon) Create Gist (GitHub)',
+    };
+    const localMarkdownOption = {
+      label: '$(markdown) Save as markdown',
+    };
+
+    // User selects export format in quick pick
+    const exportFormat = await window.showQuickPick(
+      [gistOption, localMarkdownOption],
+      {
+        placeHolder: 'Select export format',
+        canPickMany: false,
+        ignoreFocusOut: true,
+      }
+    );
+
+    if (!exportFormat || !exportFormat.label) {
+      throw new UserCancellationException('No export format selected', true);
+    }
+
+    if (exportFormat === gistOption) {
+      const description = 'CodeQL Variant Analysis Results';
+
+      const markdownFiles = generateMarkdown(query, analysesResults, 'gist');
+
+      // Convert markdownFiles to the appropriate format for uploading to gist
+      const gistFiles = markdownFiles.reduce((acc, cur) => {
+        acc[`${cur.fileName}.md`] = { content: cur.content.join('\n') };
+        return acc;
+      }, {} as { [key: string]: { content: string } });
+
+      const gistUrl = await createGist(credentials, description, gistFiles);
+      void showAndLogInformationMessage(`Variant analysis results exported to [gist](${gistUrl}).`);
```

Comment:
This will show a "Show log" button. I don't think we want that. Shall we have an "Open gist" action instead that opens the URL? (assuming that's not too hard)

Reply:
Done! Pop-up window now looks like this 
![image](https://user-images.githubusercontent.com/42641846/167882427-0b78ab27-101e-4566-92e6-7275a230dab1.png)